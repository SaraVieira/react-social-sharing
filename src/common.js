import styled, { css } from 'styled-components'
import is, { isOr } from 'styled-is'

export const SharingButton = styled.div`
  border-radius: 5px;
  transition: 25ms ease-out;
  padding: 0.5em 0.75em;
  font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
`

export const Link = styled.a`
  text-decoration: none;
  color: #fff;
  margin: 0.5em;
  display: inline-block;

  svg {
    width: 1em;
    height: 1em;
    vertical-align: top;

    &:not(:only-child) {
      margin-right: 0.4em;
    }
  }

  ${is('small')`
    svg {
      vertical-align: middle;
    }
  `};

  ${is('simple')`
    svg {
      stroke: #232323;
      fill: none;
    }
  `};

  ${is('simpleReverse')`
    svg {
      stroke: #fff;
      fill: none;
    }
  `};
`

export const Icon = styled.div`
  display: flex;
  align-items: center;
  fill: #fff;
  stroke: none;

  ${isOr('solid', 'solidcircle')`
    fill: #fff;
    stroke: none
  `};
`
