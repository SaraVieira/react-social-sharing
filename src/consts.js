export default {
  twitter: (link = '', message = '') =>
    `https://twitter.com/intent/tweet/?text=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  facebook: (link: '') =>
    `https://facebook.com/sharer/sharer.php?u=${encodeURIComponent(link)}`,
  google: (link: '') =>
    `https://plus.google.com/share?url=${encodeURIComponent(link)}`,
  tumblr: (link: '') =>
    `http://tumblr.com/widgets/share/tool?canonicalUrl=${encodeURIComponent(
      link
    )}`,
  reddit: (link: '') =>
    `https://reddit.com/submit/?url=${encodeURIComponent(link)}`,
  whatsapp: (link = '', message = '') =>
    `https://api.whatsapp.com/send?text=${encodeURIComponent(message)}%20${encodeURIComponent(
      link
    )}`,
  telegram: (link = '', message = '') =>
    `https://telegram.me/share/url?text=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  vk: (link = '', message = '') =>
    `http://vk.com/share.php?title=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  hacker: (link = '', message = '') =>
    `https://news.ycombinator.com/submitlink?u=${encodeURIComponent(
      link
    )}&t=${encodeURIComponent(message)}`,
  xing: (link = '', message = '') =>
    `https://www.xing.com/app/user?op=share;url=${encodeURIComponent(
      link
    )};title=${encodeURIComponent(message)}.`,
  mail: (link = '', subject, body) =>
    `mailto:?subject=${encodeURIComponent(
      subject || ''
    )}&body=${encodeURIComponent((body && `${body}\n\n${link}`) || link)}`,
  pinterest: (link = '', message = '') =>
    `https://pinterest.com/pin/create/button/?url=${encodeURIComponent(
      link
    )}&media=${encodeURIComponent(link)}&description=${encodeURIComponent(
      message
    )}`,
  linkedin: (link = '', message = '') =>
    `https://www.linkedin.com/shareArticle?mini=true&url=${encodeURIComponent(
      link
    )}&title=${encodeURIComponent(message)}&summary=${encodeURIComponent(
      message
    )}&source=${encodeURIComponent(link)}`
}
