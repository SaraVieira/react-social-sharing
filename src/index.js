// this should be the entry point to your library
module.exports = {
  Twitter: require('./buttons/twitter').default,
  Facebook: require('./buttons/facebook').default,
  Google: require('./buttons/google').default,
  Tumblr: require('./buttons/tumblr').default,
  Mail: require('./buttons/mail').default,
  Pinterest: require('./buttons/pinterest').default,
  Linkedin: require('./buttons/linkedin').default,
  Reddit: require('./buttons/reddit').default,
  Xing: require('./buttons/xing').default,
  Whatsapp: require('./buttons/whatsapp').default,
  HackerNews: require('./buttons/hacker').default,
  VK: require('./buttons/vk').default,
  Telegram: require('./buttons/telegram').default,
  X: require('./buttons/x').default
}
