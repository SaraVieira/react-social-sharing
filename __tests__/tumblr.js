import React from 'react'
import { Tumblr } from '../src/index'
import 'jest-styled-components'

describe('Tumblr', function() {
  test('1', () => {
    const wrapper = mount(
      <Tumblr
        solidcircle
        big
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('2', () => {
    const wrapper = mount(
      <Tumblr
        solidcircle
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('3', () => {
    const wrapper = mount(
      <Tumblr
        solidcircle
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('4', () => {
    const wrapper = mount(
      <Tumblr
        circle
        big
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('5', () => {
    const wrapper = mount(
      <Tumblr
        circle
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('6', () => {
    const wrapper = mount(
      <Tumblr
        circle
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('7', () => {
    const wrapper = mount(
      <Tumblr
        solid
        big
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('8', () => {
    const wrapper = mount(
      <Tumblr
        solid
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('9', () => {
    const wrapper = mount(
      <Tumblr
        solid
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('11', () => {
    const wrapper = mount(<Tumblr circle />)
    expect(wrapper).toMatchSnapshot()
  })
  test('11', () => {
    const wrapper = mount(<Tumblr solidcircle />)
    expect(wrapper).toMatchSnapshot()
  })
  test('12', () => {
    const wrapper = mount(<Tumblr />)
    expect(wrapper).toMatchSnapshot()
  })
  test('13', () => {
    const wrapper = mount(<Tumblr simple />)
    expect(wrapper).toMatchSnapshot()
  })
  test('14', () => {
    const wrapper = mount(<Tumblr simpleReverse />)
    expect(wrapper).toMatchSnapshot()
  })
})
