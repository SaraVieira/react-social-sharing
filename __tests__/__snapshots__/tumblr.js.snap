// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tumblr 1 1`] = `
<Button
  big={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
              <span>
                share on Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 2 1`] = `
<Button
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
              <span>
                Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 3 1`] = `
<Button
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
        small={true}
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 4 1`] = `
<Button
  big={true}
  circle={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
              <span>
                share on Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 5 1`] = `
<Button
  circle={true}
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
              <span>
                Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 6 1`] = `
<Button
  circle={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
        small={true}
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 7 1`] = `
<Button
  big={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  solid={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5.5v5h5v4h-5V15c0 5 3.5 4.4 6 2.8v4.4c-6.7 3.2-12 0-12-4.2V9.5h-3V6.7c1-.3 2.2-.7 3-1.3.5-.5 1-1.2 1.4-2 .3-.7.6-1.7.7-3h3.8z"
                />
              </svg>
               
              <span>
                share on Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 8 1`] = `
<Button
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
  solid={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5.5v5h5v4h-5V15c0 5 3.5 4.4 6 2.8v4.4c-6.7 3.2-12 0-12-4.2V9.5h-3V6.7c1-.3 2.2-.7 3-1.3.5-.5 1-1.2 1.4-2 .3-.7.6-1.7.7-3h3.8z"
                />
              </svg>
               
              <span>
                Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 9 1`] = `
<Button
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
  solid={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl=http%3A%2F%2Fsharingbuttons.io"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        message="I am so cool"
        small={true}
      >
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5.5v5h5v4h-5V15c0 5 3.5 4.4 6 2.8v4.4c-6.7 3.2-12 0-12-4.2V9.5h-3V6.7c1-.3 2.2-.7 3-1.3.5-.5 1-1.2 1.4-2 .3-.7.6-1.7.7-3h3.8z"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 11 1`] = `
<Button
  circle={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl="
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl="
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
              <span>
                Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 11 2`] = `
<Button
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl="
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl="
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                enableBackground="new 0 0 24 24"
                height="24px"
                version="1.1"
                viewBox="0 0 24 24"
                width="24px"
                x="0px"
                xmlSpace="preserve"
                y="0px"
              >
                <path
                  d="M12,0C5.383,0,0,5.383,0,12s5.383,12,12,12s12-5.383,12-12S18.617,0,12,0z M15.492,17.616C11.401,19.544,9.5,17,9.5,14.031 V9.5h-2V8.142c0.549-0.178,1.236-0.435,1.627-0.768c0.393-0.334,0.707-0.733,0.943-1.2c0.238-0.467,0.401-0.954,0.49-1.675H12.5v3h2 v2h-2v3.719c0,2.468,1.484,2.692,2.992,1.701V17.616z"
                />
              </svg>
               
              <span>
                Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 12 1`] = `
<Button>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl="
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ewcqji"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl="
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-bZQynM dkGtst"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5.5v5h5v4h-5V15c0 5 3.5 4.4 6 2.8v4.4c-6.7 3.2-12 0-12-4.2V9.5h-3V6.7c1-.3 2.2-.7 3-1.3.5-.5 1-1.2 1.4-2 .3-.7.6-1.7.7-3h3.8z"
                />
              </svg>
               
              <span>
                Tumblr
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 13 1`] = `
<Button
  simple={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl="
    rel="noreferrer noopener"
    simple={true}
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH ljNxya"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl="
      rel="noreferrer noopener"
      target="_blank"
    >
      <TumblrIconFill>
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.5.5v5h5v4h-5V15c0 5 3.5 4.4 6 2.8v4.4c-6.7 3.2-12 0-12-4.2V9.5h-3V6.7c1-.3 2.2-.7 3-1.3.5-.5 1-1.2 1.4-2 .3-.7.6-1.7.7-3h3.8z"
          />
        </svg>
      </TumblrIconFill>
    </a>
  </styled.a>
</Button>
`;

exports[`Tumblr 14 1`] = `
<Button
  simpleReverse={true}
>
  <styled.a
    aria-label="Share on Tumblr"
    href="http://tumblr.com/widgets/share/tool?canonicalUrl="
    rel="noreferrer noopener"
    simpleReverse={true}
    target="_blank"
  >
    <a
      aria-label="Share on Tumblr"
      className="sc-bwzfXH dqksVq"
      href="http://tumblr.com/widgets/share/tool?canonicalUrl="
      rel="noreferrer noopener"
      target="_blank"
    >
      <TumblrIconFill>
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.5.5v5h5v4h-5V15c0 5 3.5 4.4 6 2.8v4.4c-6.7 3.2-12 0-12-4.2V9.5h-3V6.7c1-.3 2.2-.7 3-1.3.5-.5 1-1.2 1.4-2 .3-.7.6-1.7.7-3h3.8z"
          />
        </svg>
      </TumblrIconFill>
    </a>
  </styled.a>
</Button>
`;
